src/ASM/BasisFunctionCache.C

```diff
@@ -13,6 +13,8 @@
 
 #include "BasisFunctionCache.h"
 
+#include <array>
```

Comment:
But `std::array` is also referred in the header file, so it makes more sense to include it there instead in my view.